M2/Macaulay2/packages/MonomialIntegerPrograms.m2

```diff
@@ -0,0 +1,1656 @@
+-- -*- coding: utf-8 -*-
+-- Copyright (C) 2020 Lily Silverstein and Jay White
+--
+-- This program is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU General Public License as published by
+-- the Free Software Foundation, either version 3 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU General Public License for more details.
+--
+-- You should have received a copy of the GNU General Public License
+-- along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+newPackage (
```

Comment:
Take a look at the new optional argument "OptionalComponentsPresent", which will allow you to set it up so that when "scip" is actually installed on the machine, the examples will be run. Other packages are using it, so see how they do it.

Also, check out "findProgram" and "runProgram", as new tools (since 1.16) for finding external programs.

Reply:
Looking into this now.